src/app/sidebar/aichat.tsx

```diff
@@ -0,0 +1,322 @@
+            behavior: "auto",
+            top: this.chatWindowRef.current.scrollHeight,
+        });
+    }
+
+    renderError(err: string): any {
+        return <div className="chat-msg-error">{err}</div>;
+    }
+
+    renderChatMessage(chatItem: OpenAICmdInfoChatMessageType): any {
+        const curKey = "chatmsg-" + this.chatListKeyCount;
+        this.chatListKeyCount++;
+        const senderClassName = chatItem.isassistantresponse ? "chat-msg-assistant" : "chat-msg-user";
+        const msgClassName = `chat-msg ${senderClassName}`;
+
+        let innerHTML: React.JSX.Element = (
+            <>
+                <div className="chat-msg-header">
+                    <i className="fa-sharp fa-solid fa-user"></i>
+                </div>
+                <div className="msg-text">{chatItem.userquery}</div>
+            </>
+        );
+        if (chatItem.isassistantresponse) {
+            if (chatItem.assistantresponse.error != null && chatItem.assistantresponse.error !== "") {
+                innerHTML = this.renderError(chatItem.assistantresponse.error);
+            } else {
+                innerHTML = (
+                    <>
+                        <div className="chat-msg-header">
+                            <i className="fa-sharp fa-solid fa-sparkles"></i>
+                        </div>
+                        <Markdown text={chatItem.assistantresponse.message} codeSelect />
+                    </>
+                );
+            }
+        }
+
+        const cssVar = GlobalModel.isDev ? "--app-panel-bg-color-dev" : "--app-panel-bg-color";
+        const panelBgColor = getComputedStyle(document.documentElement).getPropertyValue(cssVar).trim();
+        const color = tinycolor(panelBgColor);
+        const newColor = color.isValid() ? tinycolor(panelBgColor).darken(6).toString() : "none";
+        const backgroundColor = this.chatListKeyCount % 2 === 0 ? "none" : newColor;
+
+        return (
+            <div className={msgClassName} key={curKey} style={{ backgroundColor }}>
+                {innerHTML}
+            </div>
+        );
+    }
+
+    render() {
+        const chatMessageItems = GlobalModel.inputModel.AICmdInfoChatItems.slice();
+        const chitem: OpenAICmdInfoChatMessageType = null;
+        return (
+            <OverlayScrollbarsComponent
+                ref={this.containerRef}
+                className="content"
+                options={{ scrollbars: { autoHide: "leave" } }}
+                events={{ initialized: this.onScrollbarInitialized }}
+            >
+                <div ref={this.chatWindowRef} className="chat-window">
+                    <div className="filler"></div>
+                    <For each="chitem" index="idx" of={chatMessageItems}>
+                        {this.renderChatMessage(chitem)}
+                    </For>
+                </div>
+            </OverlayScrollbarsComponent>
+        );
+    }
+}
+
+@mobxReact.observer
+class AIChat extends React.Component<{}, {}> {
```

Comment:
Can we not reuse more from the aichat auxview or maybe make that a component somewhere else that both views can utilize? Same for the keybindings